smbios/baseboard_information.go

```diff
@@ -6,48 +6,104 @@ package smbios
 
 import "github.com/digitalocean/go-smbios/smbios"
 
-// BaseboardInformationStructure represents the SMBIOS baseboard information structure.
-type BaseboardInformationStructure struct {
-	*smbios.Structure
+// BaseboardInformation represents the SMBIOS baseboard information.
+type BaseboardInformation struct {
+	// Manufacturer returns the baseboard manufacturer.
+	Manufacturer string
+	// Product returns the baseboard product.
+	Product string
+	// Version returns the baseboard version.
+	Version string
+	// SerialNumber returns the baseboard serial number.
+	SerialNumber string
+	// AssetTag returns the baseboard asset tag.
+	AssetTag *string
+	// LocationInChassis returns the number of a null-terminated string that
+	// describes this board's location within the chassis referenced by the
+	// Chassis Handle (described below in this table)
+	// NOTE: This field supports a CIM_Container class mapping where:
+	// 	- LocationWithinContainer is this field.
+	// 	- GroupComponent is the chassis referenced by Chassis Handle.
+	// 	- PartComponent is this baseboard
+	LocationInChassis *string
+	// BoardType identifies the type of board. See 7.3.2.
+	BoardType BoardType
 }
 
-// BaseboardInformation returns a `BaseboardInformationStructure`.
-func (s *SMBIOS) BaseboardInformation() BaseboardInformationStructure {
-	return s.BaseboardInformationStructure
+// NewBaseboardInformation initializes and returns a new `BaseboardInformation`.
+func NewBaseboardInformation(s *smbios.Structure) *BaseboardInformation {
+	return &BaseboardInformation{
+		Manufacturer:      *GetString(s, 0x04),
```

Comment:
it looks like this can panic, as `GetString` might return `nil`, so we probably need to wrap it, I wonder if that's what `get` function was doing

Reply:
Yes and no. I deliberately did a few de-references which could panic.
According to the SMBIOS specification, these strings should always be present.
See Annex A of https://www.dmtf.org/sites/default/files/standards/documents/DSP0134_2.7.1.pdf (page 122).

Unfortunately it seems that not all vendors are honoring the specification, so maybe we should always default to an empty string?

Reply:
By the way, the whole `GetString()` function is there because SMBIOS uses a 2-step mechanism for retrieving strings.
Inside the structure, `s.Formatted[index]` contains a byte representing the 1-based index inside `s.Strings[]`. 
The `GetString()` function uses that byte to get the actual string from that slice (but only if its not 0).

Reply:
It is really important for `go-smbios` not to panic, I think getting empty string is fine here. 

We can add `GetStringOrEmpty` or something like that to convert to empty string if the field is not available

Reply:
Yeah, I agree it's better to be safe than sorry. I changed the function and renamed it to `GetStringOrEmpty()`